src/config.rs

```diff
@@ -0,0 +1,296 @@
+//! Config API: manage and inspect docker configs within a swarm
+
+pub use crate::models::*;
+
+use super::Docker;
+use crate::{docker::BodyType, errors::Error};
+use bytes::Bytes;
+use http::request::Builder;
+use http_body_util::Full;
+use hyper::Method;
+use serde_derive::Serialize;
+use std::{collections::HashMap, hash::Hash};
+
+/// Parameters used in the [List Config API](super::Docker::list_configs())
+///
+/// ## Examples
+///
+/// ```rust
+/// # use std::collections::HashMap;
+/// # use std::default::Default;
+/// use bollard::config::ListConfigsOptions;
+///
+/// let mut filters = HashMap::new();
+/// filters.insert("name", vec!["my-config-name"]);
+///
+/// ListConfigsOptions{
+///     filters,
+/// };
+/// ```
+///
+/// ```rust
+/// # use bollard::config::ListConfigsOptions;
+/// # use std::default::Default;
+///
+/// let options: ListConfigsOptions<&str> = Default::default();
+/// ```
+#[derive(Debug, Clone, Default, PartialEq, Serialize)]
+pub struct ListConfigsOptions<T>
+where
+    T: Into<String> + Eq + Hash + serde::ser::Serialize,
+{
+    /// Filters to process on the config list, encoded as JSON. Available filters:
+    ///  - `id`=`<ID>` a config's ID
+    ///  - `label`=`key` or `label`=`"key=value"` of a config label
+    ///  - `name`=`<name>` a config's name
+    ///  - `names`=`<name>` a multiple config's name comma separated
+    #[serde(serialize_with = "crate::docker::serialize_as_json")]
+    pub filters: HashMap<T, Vec<T>>,
+}
+
+/// Parameters used in the [Update Config API](Docker::update_config())
+///
+/// ## Examples
+///
+/// ```rust
+/// use bollard::config::UpdateConfigOptions;
+///
+/// UpdateConfigOptions{
+///     version: 1234,
+///     ..Default::default()
+/// };
+/// ```
+#[derive(Debug, Copy, Clone, Default, PartialEq, Serialize)]
+pub struct UpdateConfigOptions {
```

Comment:
Both this and the `ListConfigOptions` should get generated from the query_parameter maven swagger generation. Add a clause here for each missing query parameter: https://github.com/fussybeaver/bollard/blob/46489623180713e6b0b24a77fa305dc763c33236/codegen/swagger/pom.xml#L37 - then generate using the command detailed in the README..